d2cli/main.go

```diff
@@ -298,8 +299,16 @@ func Run(ctx context.Context, ms *xmain.State) (err error) {
 		return w.run()
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, time.Minute*2)
-	defer cancel()
+	timeout := time.Minute * 2
```

Comment:
Instead of repeating it in 3 places:

```
func WithTimeout(ctx) (ctx, cancel) {
    if env.timeout > 0 { return context.WithTimeout() }
    return ctx, func() {}
}
```

Also, shouldn't we enforce like a minimum of 60s or so? Not sure if it makes sense to have timeout=1

Reply:
created `WithTimeout` util func for this